app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt

```diff
@@ -3,69 +3,106 @@ package ch.epfl.sdp.blindly.recording
-        recordButton.check(matches(withText("Start recording")))
-        recordButton.perform(click())
-        recordButton.check(matches(withText("Stop recording")))
-        recordButton.perform(click())
-        recordButton.check(matches(withText("Start recording")))
+    fun audioDurationTextIsCorrectlyDisplayed() {
+        createRecord(200L)
+        val recordDuration = Espresso.onView(withId(R.id.recordDuration))
+        recordDuration.check(matches(withText(ZERO_SECONDS_TEXT)))
     }
 
     @Test
-    fun playingButtonIsCorrectlyDisabled() {
-        val playButton = Espresso.onView(withId(R.id.playingButton))
-        playButton.check(matches(not(isEnabled())))
+    fun recordingActivityFiresProfileFinished() {
+        createRecord(200L)
+        Espresso.onView(withId(R.id.nameDurationLayout))
+                .perform(click())
+
+        Intents.init()
+        Espresso.onView(withId(R.id.selectButton))
+                .perform(click())
+        Intents.intended(IntentMatchers.hasComponent(ProfileFinished::class.java.name))
+        Intents.release()
     }
 
     @Test
-    fun playingButtonIsCorrectlyEnabled() {
-        val recordButton = Espresso.onView(withId(R.id.recordingButton))
-        val playButton = Espresso.onView(withId(R.id.playingButton))
-        recordButton.perform(click(), click())
-        playButton.check(matches(isEnabled()))
+    fun playPauseButtonChangesBackgroundWhenClickedOnce() {
+        createRecord(500L)
+        Espresso.onView(withId(R.id.nameDurationLayout))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .check(
+                        matches(
+                                withDrawable(android.R.drawable.ic_media_pause)
+                        )
+                )
     }
 
     @Test
-    fun playingButtonTextIsCorrectlyDisplayed() {
+    fun playPauseButtonChangesBackgroundWhenClickedTwice() {
+        createRecord(500L)
+        Espresso.onView(withId(R.id.nameDurationLayout))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .check(
+                        matches(
+                                withDrawable(android.R.drawable.ic_media_play)
+                        )
+                )
+    }
+
+    @Test
+    fun playPauseButtonChangesBackgroundWhenPlayIsFinished() {
+        createRecord(500L)
+        Espresso.onView(withId(R.id.nameDurationLayout))
+                .perform(click())
+        Espresso.onView(withId(R.id.playPauseButton))
+                .perform(click())
+        Thread.sleep(2000L)
+        Espresso.onView(withId(R.id.playPauseButton))
```

Comment:
It might be better to import Espreso.onView in this test class, since it's called everywhere